app/models/group.rb

```diff
@@ -1,12 +1,27 @@
 # frozen_string_literal: true
 
 class Group < ApplicationRecord
-  groupify :group, default_members: :users
+  has_many :group_memberships
+  has_many :users, through: :group_memberships
+
   validates :name, presence: true
 
   scope :discoverable, -> { where(visible: true) }
+  scope :as, lambda { |membership_type|
+    joins(:group_memberships).where(
+      group_memberships: { membership_type: membership_type }
+    )
+  }
 
   def add_owner(user)
-    add user, as: :owner
+    user.group_memberships.create(membership_type: :owner, group: self)
+  end
+
+  def add(user)
+    user.group_memberships.create(group: self)
```

Comment:
unless you are going to check for the return status (doubt), I'd go with `create!`

Reply:
`group_memberships.create!(user: user)`